### PR TITLE
pessimistic-txn: eliminate `CleanUpWaitFor` when waiter timeouts

### PR DESCRIPTION
**Pull request by youjiali1995**
Signed-off-by: youjiali1995 <zlwgx1023@gmail.com>

## What have you changed? (mandatory)
Use `wait-for-lock-timeout` as the TTL of the detect table. So when waiter timeouts, there is no need to send 'CleanUpWaitFor` request to the deadlock detector.

## What are the type of the changes? (mandatory)
- Improvement (change which is an improvement to an existing feature)

### CODE REVIEW

**Comment by youjiali1995** [comment]
@coocood @MyonKeminta @AndreMouche PTAL

**Comment by coocood** [comment]
LGTM

File: src/binutil/server.rs [hunk 1]
```diff
@@ -322,6 +322,7 @@ fn run_raft_server(pd_client: RpcClient, cfg: &TiKvConfig, security_mgr: Arc<Sec
             pd_client,
             resolver,
             cfg.pessimistic_txn.monitor_membership_interval,
+            cfg.pessimistic_txn.wait_for_lock_timeout,
```
Comment (MyonKeminta) [inline src/binutil/server.rs]: I'm afraid there might be too many arguments. May it be better if we pass the whole `cfg.pessimistic_txn`?

File: src/storage/lock_manager/deadlock.rs [hunk 2]
```diff
@@ -310,13 +310,15 @@ impl Inner {
...
         security_mgr: Arc<SecurityManager>,
+        ttl: u64,
     ) -> Self {
         Self {
             store_id,
             leader_info: None,
             leader_client: None,
-            // TODO: make it configurable.
-            detect_table: Rc::new(RefCell::new(DetectTable::new(time::Duration::from_secs(3)))),
+            detect_table: Rc::new(RefCell::new(DetectTable::new(time::Duration::from_millis(
```
Comment (AndreMouche) [inline src/storage/lock_manager/deadlock.rs]: I think we can remove the Rc in another PR.

File: src/binutil/server.rs (hunk 1 above)
Comment (youjiali1995) [reply src/binutil/server.rs]: Yes. I will file a PR to pass `&cfg.pessimistic_txn` to `WaiterManger` and `Detector`.

**Pull request merged**